File: src/main/java/com/campus/core/entity/AuditLog.java
package com.campus.core.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 审计日志实体类
 * 用于记录系统操作审计信息
 */
@Data
public class AuditLog {

    /**
     * 日志ID
     */
    private Long id;

    /**
     * 操作用户ID
     */
    private Long userId;

    /**
     * 目标用户ID
     */
    private Long targetUserId;

    /**
     * 操作用户名
     */
    private String username;

    /**
     * 操作类型
     */
    private String operation;

    /**
     * 资源类型
     */
    private String resourceType;

    /**
     * 资源ID
     */
    private Long resourceId;

    /**
     * 请求方法
     */
    private String requestMethod;

    /**
     * 请求路径
     */
    private String requestPath;

    /**
     * 请求参数
     */
    private String requestParams;

    /**
     * 响应状态码
     */
    private Integer responseStatus;

    /**
     * 响应消息
     */
    private String responseMessage;

    /**
     * 客户端IP
     */
    private String clientIp;

    /**
     * User-Agent
     */
    private String userAgent;

    /**
     * 执行时间(ms)
     */
    private Integer executionTime;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;
}

File: web/src/api/auditLog.ts
import apiClient from './client';
import type { ApiResponse } from '@/types/common';

/** 审计日志接口 */
export interface AuditLog {
  /** 日志ID */
  id: number;
  /** 操作用户ID */
  userId: number;
  /** 目标用户ID */
  targetUserId?: number;
  /** 操作用户名 */
  username: string;
  /** 操作类型 */
  operation: string;
  /** 资源类型 */
  resourceType: string;
  /** 资源ID */
  resourceId: string;
  /** 请求方法 */
  requestMethod?: string;
  /** 请求路径 */
  requestPath?: string;
  /** 请求参数 */
  requestParams?: string;
  /** 响应状态码 */
  responseStatus?: number;
  /** 操作详情（响应消息） */
  responseMessage: string;
  /** 客户端IP地址 */
  clientIp: string;
  /** User-Agent */
  userAgent?: string;
  /** 执行时间(ms) */
  executionTime?: number;
  /** 创建时间 */
  createdAt: string;
}

/** 获取审计日志列表 */
export const getAuditLogs = (params: {
  /** 用户ID */
  userId?: number;
  /** 操作类型 */
  operation?: string;
  /** 资源类型 */
  resourceType?: string;
  /** 开始时间 */
  startTime?: string;
  /** 结束时间 */
  endTime?: string;
  /** 页码 */
  page?: number;
  /** 每页数量 */
  size?: number;
}) =>
  apiClient.get<ApiResponse<any>>('/audit-logs', { params });

/** 获取最近审计日志 */
export const getRecentAuditLogs = (limit: number = 20) =>
  apiClient.get<ApiResponse<AuditLog[]>>('/audit-logs/recent', { params: { limit } });

/** 获取我的审计日志 */
export const getMyAuditLogs = () =>
  apiClient.get<ApiResponse<AuditLog[]>>('/audit-logs/my');

/** 获取审计日志统计 */
export const getAuditLogStats = () =>
  apiClient.get<ApiResponse<any>>('/audit-logs/stats');
